### PR TITLE
CHI-2680 Refactored masking identifiers

### PR DESCRIPTION
**Pull request by mythilytm**
## Description
- This PR overhauls the masking identifiers logic and how the twilio strings that have PII is changed.
- `maskManagerStringsWithIdentifiers` is introduced that will dynamically change all the manager strings with `defaultFrom` which includes the identifier. This helps eliminate the need for hardcoding which was the practice before. Now, as long as the permission is set to mask identifiers and there is a MaskIdentifiers string, this will bubble all the twilio strings to be masked.
- All the masking Identifiers logic is now moved to the `maskIdentifiers` directory, cleaning up much of the hard to manage code with in HrmPluginForm file

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Other Related Issues

None

### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

File: plugin-hrm-form/src/HrmFormPlugin.tsx [hunk 1]
```diff
@@ -62,12 +57,20 @@ const setUpLocalization = (config: ReturnType<typeof getHrmConfig>) => {
   const { counselorLanguage, helplineLanguage } = config;
 
   const twilioStrings = { ...manager.strings }; // save the originals
-  const setNewStrings = (newStrings: typeof getTemplateStrings) =>
-    (manager.strings = { ...manager.strings, ...newStrings });
+
+  const setNewStrings = (newStrings: { [key: string]: string }) => {
+    // manager.strings = { ...manager.strings, ...newStrings };
+    // const maskedTwilioStrings = maskManagerStringsWithIdentifiers(newStrings);
```
Comment (GPaoloni) [inline plugin-hrm-form/src/HrmFormPlugin.tsx:63]: delete comments? :P

File: plugin-hrm-form/src/maskIdentifiers/index.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  // Task list and panel when a call comes in
+  TaskListItem.firstLine = 'MaskIdentifiers';
+
+  // Task panel during an active call
+  TaskCanvasHeader.title = 'MaskIdentifiers';
+  Supervisor.TaskCanvasHeader.title = 'MaskIdentifiers';
+
+  // Task Status in Agents page
+  if (!getAseloFeatureFlags().enable_teams_view_enhancements) TaskCard.firstLine = 'MaskIdentifiers';
```
Comment (GPaoloni) [inline plugin-hrm-form/src/maskIdentifiers/index.ts:49]: Is the logic here correct? "if not enable_teams_view_enhancements then set TaskCard.firstLine"?

File: plugin-hrm-form/src/maskIdentifiers/unmaskPhoneNumber/index.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  TaskCanvasHeader.Content.add(<ViewTaskNumber key="view-task-number" />, {
+    sortOrder: 1,
+    if: props => props.task.channelType === 'voice',
+  });
+
+  // Supervisor.TaskCanvasHeader.Content.add(<ViewTaskNumber key="view-supervisor-task-number" />, {
+  //   sortOrder: 0,
+  //   if: props => props.task.channelType === 'voice',
+  // });
```
Comment (GPaoloni) [inline plugin-hrm-form/src/maskIdentifiers/unmaskPhoneNumber/index.tsx:41]: Delete?

File: plugin-hrm-form/src/permissions/dev.json [hunk 4]
```diff
@@ -34,5 +34,5 @@
 
   "viewPostSurvey": [["isSupervisor"]],
 
-  "viewIdentifiers": [["everyone"]]
+  "viewIdentifiers": []
```
Comment (GPaoloni) [inline plugin-hrm-form/src/permissions/dev.json:37]: Intentional?

**Review by GPaoloni — APPROVED** [review]
LGTM, this is a great refactor, thank you!!

File: plugin-hrm-form/src/maskIdentifiers/index.ts (hunk 2 above)
Comment (mythilytm) [reply plugin-hrm-form/src/maskIdentifiers/index.ts:49]: yes, this is part of a teams view feature. We don't want to know the masked identifier if it is of no use. 
see https://tech-matters.slack.com/archives/C035K8919U7/p1712255294131169

File: plugin-hrm-form/src/permissions/dev.json (hunk 4 above)
Comment (mythilytm) [reply plugin-hrm-form/src/permissions/dev.json:37]: good catch

**Pull request merged**